synapse/http/federation/matrix_federation_agent.py

```diff
@@ -72,6 +76,7 @@ def __init__(
         tls_client_options_factory: Optional[FederationPolicyForHTTPS],
         user_agent: bytes,
         ip_blacklist: IPSet,
+        proxy_reactor: Optional[ISynapseReactor] = None,
```

Comment:
please add new params to the docstring (on the class, in this case). As well as making the code maintainable in future, it's important to help with review.